### ISSUE TITLE
Cannot pass principal to Role from '@alicloud/ros-cdk-ram' typescript package

### ISSUE THREAD

**Issue by kucharzyk-sebastian** [aliyun/Resource-Orchestration-Service-Cloud-Development-Kit#39]
Hi,

I've just noticed that it's not possible to set a principal when creating a RAM role which should be possible according to [RAM documentation](https://www.alibabacloud.com/help/en/doc-detail/48369.htm#section-4bl-sio-jkd).

I tried passing the `principal` prop but it's not defined in the interface [here](https://github.com/aliyun/Resource-Orchestration-Service-Cloud-Development-Kit/blob/master/multiple-languages/js/alicloud-ros-cdk-ram-1.0.4/lib/ram.generated.d.ts#L276) and I believe it should.

Here's a sample code I tried to run:

```typescript
import * as aliyun_ram from '@alicloud/ros-cdk-ram';

new aliyun_ram.Role(
            this.stack.aliyunStack,
            "ApiRole",
            {
                roleName: "myRole",
                description: "myRole description",
                assumeRolePolicyDocument: {
                    version: "1",
                    statement: [
                        {
                            action: ["sts:AssumeRole"],
                            effect: "Allow",
                            principal: {
                                Service: [
                                  "apigateway.aliyuncs.com"
                                ]
                            }
                        }
                    ],
                },
                policies: []
            }
        )
```

**Comment by baichenxu3** [comment]
First of all this property is supported, possibly due to the wrong action property you set that ultimately becomes unusable, For example：
import * as ros from '@alicloud/ros-cdk-core';
import * as ram from '@alicloud/ros-cdk-ram';

export class DemoStack extends ros.Stack {
  constructor(scope: ros.Construct, id: string, props?: ros.StackProps) {
    super(scope, id, props);
    new ros.RosInfo(this, ros.RosInfo.description, "This is the simple ros cdk app example.");
    // The code that defines your stack goes here
    new ram.Role(
      this, 
      'ram-from-ros-cdk', 
      {
        "roleName": "myRole",
        "description": "myRole description",
        "assumeRolePolicyDocument": {
          "version": "1",
          "statement": [
            {
              "action": "sts:AssumeRole",
              "effect": "Allow",
              "principal": {
                "service": [
                  "apigateway.aliyuncs.com"
                ]
              }
            }
          ]
        }
      }
    );
  }
}

**Comment by baichenxu3** [comment]
First of all this property is supported, possibly due to the wrong action property you set that ultimately becomes unusable, For example：
`
import * as ros from '@alicloud/ros-cdk-core';
import * as ram from '@alicloud/ros-cdk-ram';

export class DemoStack extends ros.Stack {
  constructor(scope: ros.Construct, id: string, props?: ros.StackProps) {
    super(scope, id, props);
    new ros.RosInfo(this, ros.RosInfo.description, "This is the simple ros cdk app example.");
    // The code that defines your stack goes here
    new ram.Role(
      this, 
      'ram-from-ros-cdk', 
      {
        "roleName": "myRole",
        "description": "myRole description",
        "assumeRolePolicyDocument": {
          "version": "1",
          "statement": [
            {
              "action": "sts:AssumeRole",
              "effect": "Allow",
              "principal": {
                "service": [
                  "apigateway.aliyuncs.com"
                ]
              }
            }
          ]
        }
      }
    );
  }
}
`

**Comment by baichenxu3** [comment]
First of all this property is supported, possibly due to the wrong action property you set that ultimately becomes unusable, For example：
`import * as ros from '@alicloud/ros-cdk-core';
import * as ram from '@alicloud/ros-cdk-ram';

export class DemoStack extends ros.Stack {
  constructor(scope: ros.Construct, id: string, props?: ros.StackProps) {
    super(scope, id, props);
    new ros.RosInfo(this, ros.RosInfo.description, "This is the simple ros cdk app example.");
    // The code that defines your stack goes here
    new ram.Role(
      this, 
      'ram-from-ros-cdk', 
      {
        "roleName": "myRole",
        "description": "myRole description",
        "assumeRolePolicyDocument": {
          "version": "1",
          "statement": [
            {
              "action": "sts:AssumeRole",
              "effect": "Allow",
              "principal": {
                "service": [
                  "apigateway.aliyuncs.com"
                ]
              }
            }
          ]
        }
      }
    );
  }
}`

**Comment by baichenxu3** [comment]
First of all this property is supported, possibly due to the wrong action property you set that ultimately becomes unusable, For example：
```typescript
import * as ros from '@alicloud/ros-cdk-core';
import * as ram from '@alicloud/ros-cdk-ram';

export class DemoStack extends ros.Stack {
  constructor(scope: ros.Construct, id: string, props?: ros.StackProps) {
    super(scope, id, props);
    new ros.RosInfo(this, ros.RosInfo.description, "This is the simple ros cdk app example.");
    // The code that defines your stack goes here
    new ram.Role(
      this, 
      'ram-from-ros-cdk', 
      {
        "roleName": "myRole",
        "description": "myRole description",
        "assumeRolePolicyDocument": {
          "version": "1",
          "statement": [
            {
              "action": "sts:AssumeRole",
              "effect": "Allow",
              "principal": {
                "service": [
                  "apigateway.aliyuncs.com"
                ]
              }
            }
          ]
        }
      }
    );
  }
}
```

**Comment by Ehzyo** [comment]
- It is a bug in early release of ros-cdk. But it has been fixed since 1.0.2(@alicloud/ros-cdk-core, @alicloud/ros-cdk-ram).
- I did a test as below, and it works fine.
- If you want to upgrade ros-cdk package(@alicloud/ros-cdk-core, @alicloud/ros-cdk-ram) from versions below 1.0.2 to versions 1.0.2 and above, Please test, there are some incompatible changes.
![image](https://user-images.githubusercontent.com/11574420/159394699-3c2b1baa-b6a0-41fd-bac7-3a6170eb547d.png)
![image](https://user-images.githubusercontent.com/11574420/159394298-283b60ea-d0b3-47e4-81a5-549f25068ed4.png)
![image](https://user-images.githubusercontent.com/11574420/159394419-ba00f9a8-8eec-4c04-8e9a-292caeb772b5.png)

**Comment by kucharzyk-sebastian** [comment]
Thanks, I was indeed using v1.0.1. A newer version helped!
